contracts/transfer/src/state.rs

```diff
@@ -53,136 +46,115 @@ impl TransferState {
             nullifiers: BTreeSet::new(),
             roots: ConstGenericRingBuffer::new(),
             balances: BTreeMap::new(),
-            var_crossover: None,
-            var_crossover_addr: None,
+            deposit: None,
             gas_price: None,
         }
     }
 
+    /// Mint a new phoenix note.
+    ///
+    /// This can only be called by the transfer- and stake-contracts.
+    /// If called by the `stake-contract`, this method will increase the total
+    /// amount of circulating dusk. This happens when the reward for staking
+    /// and participating in the consensus is withdrawn.
+    /// If called by the transfer-contract itself, it is important to make sure
+    /// that the minted value is subtracted from a contracts balance before
+    /// creating a phoenix-note.
```

Comment:
It is technically not true that this can be "called" by the transfer contract. i.e. an inter-contract call from the transfer contract will still fail.

Reply:
Ahh, interesting! Will adjust the comment!